packages/expo-updates/ios/EXUpdates/Logging/UpdatesLogReader.swift

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+// Reads expo-updates logs
+
+import Foundation
+import OSLog
+
+import ExpoModulesCore
+import SafariServices
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+@objc(EXUpdatesLogReader)
+public class UpdatesLogReader: NSObject {
+
+  @objc(getLogEntriesNewerThan:)
+  public func getLogEntries(newerThan: Date) -> NSArray {
+    return getLogEntries(newerThan: newerThan, returnEntriesAs: "NSDictionary")
+  }
+
+  /**
+   Get expo-updates logs newer than the given date
+   Returns an Objective-C NSArray of either strings or NSDictionary objects
+   Returned strings are all in the JSON format of UpdatesLogEntry
+   Maximum of one day lookback is allowed
+   */
+  @objc(getLogEntriesNewerThan:returnEntriesAs:)
+  public func getLogEntries(newerThan: Date, returnEntriesAs: String = "NSDictionary") -> NSArray {
```

Comment:
Prefer two methods that have more concrete return types than a dynamic return type

Reply:
Done